astropy/io/fits/tests/test_convenience.py

```diff
@@ -153,3 +153,23 @@ def test_tabledump(self):
         # test with datafile
         fits.tabledump(filename, datafile=self.temp('test_tb.txt'))
         assert os.path.isfile(self.temp('test_tb.txt'))
+
+        # test with cdfile, hfile
```

Comment:
This should go in a new test function, it makes it easier to identify failures.
And you will probably need to rebase on master since the test above was modified recently.